### PR TITLE
Mask LED events to fix QubesOS issue 5823

### PR DESCRIPTION
**Pull request by m-v-b**
Hi,

This pull request is intended to fix QubesOS/qubes-issues#5823. I have been using this patch on my personal computer for a few months, in conjunction with an up-to-date installation of QubesOS v4.2. For detailed information, please see the commit message.

Thanks!

Vefa

### CODE REVIEW

**Comment by marmarek** [comment]
It works!

File: src/input-proxy-sender.c [hunk 1]
```diff
@@ -157,6 +177,9 @@ int main(int argc, char **argv) {
 
     /* TODO: produce synthetic EV_REP initial events for keyboard */
 
+    if (ignore_led_events(fd) == -1)
+        return 1;
```
Comment (DemiMarie) [inline src/input-proxy-sender.c:181]:
```suggestion
    // The sender is a client of the evdev driver and has an active grab.
    // By default, Linux will send LED events back to us, even if we sent
    // them. This causes the events to recirculate forever, consuming
    // lots of CPU and rendering the input device unusable.
    if (ignore_led_events(fd) == -1)
        return 1;
```

**Review by marmarek — APPROVED** [review]

**Review by DemiMarie — COMMENTED** [review]
Thanks for the fix!  It would be great to summarize the commit message in a comment, like the one I suggested.  This reduces the risk of someone removing this code because they don’t realize what it does.

File: src/input-proxy-sender.c (hunk 1 above)
Comment (m-v-b) [reply src/input-proxy-sender.c:181]: Hi @DemiMarie, sorry for the delayed response. I can open another pull request to address this, if desired. Please let me know -- thanks!

Comment (DemiMarie) [reply src/input-proxy-sender.c:181]: That would be great, thank you!  It’s my fault for being so late to review.

**Pull request merged**